tests/lib/nested.sh

```diff
@@ -1230,6 +1225,60 @@ nested_exec_as() {
     sshpass -p "$PASSWD" ssh -p "$NESTED_SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$USER"@localhost "$@"
 }
 
+nested_prepare_tools() {
+    if ! nested_exec "test -e retry" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/retry"
+        nested_exec "sed 's/any-python/python3/g' -i ./retry"
+    fi
+
+    if ! nested_exec "test -e not" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/not"
+    fi
+
+    if ! nested_exec "test -e MATCH" &>/dev/null; then
+        cat >> MATCH << 'EOF'
+#!/bin/bash
+set +xu
+stdin=$(cat)
+if ! echo "$stdin" | grep -q -E "$@"; then
+    res=$?
+    echo -e "grep error: pattern not found, got:\n$stdin" >&2
+    if [ "$res" != 1 ]; then
+        echo "unexpected grep exit status: $res"
+    fi
+    exit 1
+fi
+EOF
+        chmod +x MATCH
+        nested_copy "MATCH"
+        rm -f MATCH
+    fi
+
+    if ! nested_exec "test -e NOMATCH" &>/dev/null; then
+        cat >> NOMATCH << 'EOF'
+#!/bin/bash
+set +xu
+stdin=$(cat)
+if echo "$stdin" | grep -q -E "$@"; then
+    res=$?
+    echo -e "grep error: pattern found, got:\n$stdin" >&2
+    if [ "$res" != 1 ]; then
+        echo "unexpected grep exit status: $res"
+    fi
+    exit 1
+fi
+EOF
+        chmod +x NOMATCH
+        nested_copy "NOMATCH"
+        rm -f NOMATCH
+    fi
+
+    if ! nested_exec "grep -qE PATH=.*/home/user1 /etc/environment"; then
```

Comment:
what's this for ?

Reply:
The idea of this is to avoid updating twice the path

Reply:
no sorry my point is why do we need /home/user1 in $PATH ?

Reply:
because there we store MATCH, retry and the other tools
so then we can use it

Reply:
the alternative is to copy them to another place which is already in path, any suggestion?

Reply:
ah right that makes sense, in this case storing in /home/user1 and $PATH is fine